tutorials/dapt-curation/code/utils.py

```diff
@@ -0,0 +1,389 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# -
+
+import json
+import re
+
+import dask.dataframe as dd
+import pandas as pd
+from modifiers import QuotationUnifier
+
+from nemo_curator import ScoreFilter, Sequential
+from nemo_curator.datasets import DocumentDataset
+from nemo_curator.filters import (
+    DocumentFilter,
+    RepeatedLinesFilter,
+    RepeatedParagraphsFilter,
+    RepeatingTopNGramsFilter,
+    UrlsFilter,
+    WordCountFilter,
+)
+from nemo_curator.filters.code import (
+    NumberOfLinesOfCodeFilter,
+    PythonCommentToCodeFilter,
+)
+from nemo_curator.modifiers import DocumentModifier
+from nemo_curator.modifiers.pii_modifier import PiiModifier
+from nemo_curator.modifiers.unicode_reformatter import UnicodeReformatter
+from nemo_curator.modules import ExactDuplicates
+from nemo_curator.modules.modify import Modify
+from nemo_curator.pii.constants import DEFAULT_LANGUAGE, DEFAULT_MAX_DOC_SIZE
+from nemo_curator.utils.distributed_utils import get_client
+from nemo_curator.utils.file_utils import get_all_files_paths_under
+from nemo_curator.utils.script_utils import add_distributed_args
+
+
+class QuotationUnifier(DocumentModifier):
+    """
+    A simple modifier that unifies the quotation marks in the documents.
+    """
+
+    def modify_document(self, text: str) -> str:
+        """
+        Modifies the given text by replacing left and right single quotes with normal single quotes,
+        and replacing left and right double quotes with normal double quotes.
+
+        Args:
+            text (str): The text to be modified.
+
+        Returns:
+            str: The modified text.
+        """
+        text = text.replace("‘", "'").replace("’", "'")
+        text = text.replace("“", '"').replace("”", '"')
+        return text
+
+
+def clean_and_unify(dataset: DocumentDataset) -> DocumentDataset:
+    """
+    Cleans and unifies the given dataset using a set of predefined cleaners.
+
+    Args:
+        dataset (DocumentDataset): The dataset to be cleaned and unified.
+
+    Returns:
+        DocumentDataset: The cleaned and unified dataset.
+    """
+    cleaners = Sequential(
+        [
+            # Unify all the quotation marks
+            Modify(QuotationUnifier(), text_field="text"),
+            # Unify all unicode
+            Modify(UnicodeReformatter(), text_field="text"),
+        ]
+    )
+    return cleaners(dataset)
+
+
+def filter_txt(dataset: DocumentDataset) -> DocumentDataset:
```

Comment:
[nit] rename to `filter_text_sources` so that it's more descriptive?

Reply:
We have filter_code and filter_txt. I think that's descriptive enough?

Reply:
Yes, but let's rename to `filter_text`.

Reply:
done